test/e2e/e2ethanos/services.go

```diff
@@ -313,3 +313,33 @@ func NewStoreGW(sharedDir string, name string, bucketConfig client.BucketConfig,
 		9091,
 	), nil
 }
+
+func NewCompactor(sharedDir string, name string, bucketConfig client.BucketConfig, extArgs ...string) (*Service, error) {
+	dir := filepath.Join(sharedDir, "data", "compact", name)
+	container := filepath.Join(e2e.ContainerSharedDir, "data", "compact", name)
+
+	if err := os.MkdirAll(dir, 0777); err != nil {
+		return nil, errors.Wrap(err, "create compact dir")
+	}
+
+	bktConfigBytes, err := yaml.Marshal(bucketConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "generate compact config file: %v", bucketConfig)
+	}
+
+	return NewService(
+		fmt.Sprintf("compact-%s", name),
+		DefaultImage(),
+		e2e.NewCommand("compact", append(e2e.BuildArgs(map[string]string{
+			"--debug.name":      fmt.Sprintf("compact-%s", name),
+			"--log.level":       logLevel,
+			"--data-dir":        container,
+			"--objstore.config": string(bktConfigBytes),
+			"--http-address":    ":80",
+			"--wait":            "",
```

Comment:
It would be nice to increase loop internval for Compator for test purposes.

Reply:
👍  https://github.com/thanos-io/thanos/pull/2265